go-controller/pkg/ovn/address_set/address_set.go

```diff
@@ -91,6 +95,16 @@ func (asf *ovnAddressSetFactory) NewAddressSet(dbIDs *libovsdbops.DbObjectIDs, i
 	return asf.ensureOvnAddressSets(ips, dbIDs, true)
 }
 
+// NewAddressSetOps returns a new object that implements AddressSet
+// and contains the given IPs, or an error. Internally it creates
+// address set ops for IPv4 and IPv6 each.
+func (asf *ovnAddressSetFactory) NewAddressSetOps(dbIDs *libovsdbops.DbObjectIDs, ips []net.IP) (AddressSet, []ovsdb.Operation, error) {
```

Comment:
yeah i wanted to do that as well but didn't do it because because of this weird match we do here:
```
	// UUID should always be set if no error, check anyway
	if err != nil || addrSet.UUID == "" {
		// NOTE: While ovsdb transactions get serialized by libovsdb, the decision to create vs. update
		// the address set takes place before that serialization is done. Because of that, it is feasible
		// that one of the go threads attempting to call this routine at the same time will fail.
		// This is described in https://bugzilla.redhat.com/show_bug.cgi?id=2108026 . While we could
		// handle that failure here by retrying, a higher level retry (see retry_obj.go) is already
		// present in the codepath, so no additional handling for that condition has been added here.
		return nil, fmt.Errorf("failed to create or update address set %+v: %w", addrSet, err)
	}
```
I will not have an `addrSet.UUID` set for my ops part because its not created yet... but maybe that check is not really needed. I have no idea behind the history of this but as of now i don't see any reason to check specifically for the UUID.

Reply:
changed `EnsureAddressSet` and `NewAddressSet` to use `ensureOvnAddressSetsOps` instead and transact it.